### PR TITLE
Added Edit, delete photo journey and improved the interface

### PR DESCRIPTION
**Pull request by eraulo**
- User can edit photo and user can delete photo, if user is active or staff
- Improve the interface

### CODE REVIEW

**Comment by PeteCoward** [comment]
This contains commits from other branches that you have open PRs from.
This PR should merge into  one of those branches.

When you create a branch with `git checkout -b newbranchname` you are branching from the branch that you are on. Your pull request should go into that branch.

If you want to branch from master, the either `git checkout master` then `git checkout -b newbranchname` OR `git checkout -b newbranchname origin/master`

File: map/templates/map/updatephototimor_form.html [hunk 1]
```diff
@@ -0,0 +1,98 @@
+
+{% extends 'map/base.html' %}
+{% load i18n %}
+{% load widget_tweaks %}
+
+{% block title %}{% trans 'Add journey photos' %}{% endblock %}
```
Comment (PeteCoward) [inline map/templates/map/updatephototimor_form.html:6]: this title is from the add form

File: map/urls.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
     path('another', AnotherView.as_view()),
     path('hatama_viazen/', HatamaViazenView.as_view(), name='hatama_viazen'),
     path('<int:pk>/update', ViazenUpdateView.as_view(), name='update_viazen'),
+    path('<int:pk>/update_photo', UpdatePhotoViazenView.as_view(), name='update_photo'),
+    path('<int:pk>/delete_photo', DeletePhotoView.as_view(), name='delete_photo'),
     path('<int:pk>/photo_viazen/', PhotoViazenView.as_view(), name='photo_viazen'),
     path('<int:pk>/delete', ViazenDeleteView.as_view(), name="delete_viazen"),
```
Comment (PeteCoward) [inline map/urls.py:16]:
```suggestion
    path('viazen/new', HatamaViazenView.as_view(), name='hatama_viazen'),
    path('viazen/update/<int:pk>', ViazenUpdateView.as_view(), name='update_viazen'),
    path('viazen/delete/<int:pk>', ViazenDeleteView.as_view(), name="delete_viazen"),
    path('photo/update/<int:pk>', UpdatePhotoViazenView.as_view(), name='update_photo'),
    path('photo/delete/<int:pk>', DeletePhotoView.as_view(), name='delete_photo'),
    path('viazen/<int:pk>/photo/new/', PhotoViazenView.as_view(), name='photo_viazen'),
 ```

File: map/static/css/timor.css [hunk 3]
```diff
@@ -144,6 +144,7 @@ button.btn.btn-success.btn-lg {
 img.gallery-img {
     max-width: 100%;
     height: auto;
+    cursor: pointer;
```
Comment (PeteCoward) [inline map/static/css/timor.css:147]: only do this to things which can be clicked!

File: map/templates/map/phototimor_form.html [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
 {% block content %}
+ <style>
+    label {
+        text-indent: -9999px;
+    }
+    #id_istoriaviazen {
+        text-indent: -9999px;
+        width: 0px;
+    }
+</style>
```
Comment (PeteCoward) [inline map/templates/map/phototimor_form.html:13]: inline style, why use inline style, why not add to timor.css?

File: map/templates/map/mapview.html [hunk 5]
```diff
@@ -51,7 +51,33 @@ <h3 class="trip-text">{% trans 'Our histories for our trip' %}</h3>
                             {%for photo in geoimages %}
                                 {% if photo.viazen_id == via.id %}
                                     <figure>
-                                        <img src='{{ photo.photo }}' class="gallery-img" width="300" height="240">
+                                        <img src='{{ photo.photo }}' class="gallery-img" width="300" height="240" alt="Image {{ photo.photo_id }}" data-toggle="modal" data-target="#{{ photo.photo_id }}">
+                                        <!-- Modal -->
+                                        <div class="modal fade" id="{{ photo.photo_id }}" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
```
Comment (PeteCoward) [inline map/templates/map/mapview.html:56]:
```suggestion
                                        <div class="modal fade" id="photo-{{ photo.photo_id }}" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
```

File: map/templates/map/mapview.html [hunk 6]
```diff
@@ -51,7 +51,33 @@ <h3 class="trip-text">{% trans 'Our histories for our trip' %}</h3>
                             {%for photo in geoimages %}
                                 {% if photo.viazen_id == via.id %}
                                     <figure>
-                                        <img src='{{ photo.photo }}' class="gallery-img" width="300" height="240">
+                                        <img src='{{ photo.photo }}' class="gallery-img" width="300" height="240" alt="Image {{ photo.photo_id }}" data-toggle="modal" data-target="#{{ photo.photo_id }}">
```
Comment (PeteCoward) [inline map/templates/map/mapview.html:54]:
```suggestion
                                        <img src='{{ photo.photo }}' class="gallery-img" width="300" height="240" alt="Image {{ photo.photo_id }}" data-toggle="modal" data-target="#photo-{{ photo.photo_id }}">
```

File: map/templates/map/phototimor_form.html (hunk 4 above)
Comment (marobo) [reply map/templates/map/phototimor_form.html:13]: tenki muda ba timor.css

File: map/templates/map/updatephototimor_form.html [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+{% block content %}
+<style>
+    label {
+        text-indent: -9999px;
+    }
+    #id_istoriaviazen {
+        text-indent: -9999px;
+        width: 0px;
+    }
+</style>
```
Comment (marobo) [inline map/templates/map/updatephototimor_form.html:17]: muda ba timor.css

File: map/templates/map/updatephototimor_form.html [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+            </div>
+        </div>
+    </div>
+  </div>
+</div>
+<script>
+    document.getElementsByClassName("btn")[0].click();
+    $('option').hide();
+</script>
+{% endblock %}
```
Comment (marobo) [inline map/templates/map/updatephototimor_form.html:109]: indentation kahur malu, 2 space ho four sapce,
hotu-hotu tenki four space

**Comment by eraulo** [comment]
Kolega sira branch ida ne'e hau merge ona ba master tamba maun Peter hare laiha problema source code. Karik style foun mai ita sei muda hotu. Maibe branch sementara hau merge ona ba master. Branch ne'e halo maun Peter review baruk ona tamba kleur loos ona.

**Pull request merged**